compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -758,6 +756,11 @@ void buildLLVMCPUCodegenPassPipeline(OpPassManager &passManager) {
   passManager.nest<ModuleOp>().addNestedPass<func::FuncOp>(
       createLLVMCPUMaterializeEncodingPass());
   passManager.addNestedPass<ModuleOp>(createBufferizeCopyOnlyDispatchesPass());
+  passManager.nest<ModuleOp>().addNestedPass<func::FuncOp>(
+      IREE::LinalgExt::createDecomposeSoftmaxPass());
+  // Temporary solution to avoid large allocations due to softmax lowering.
+  passManager.nest<ModuleOp>().addNestedPass<func::FuncOp>(
+      createRematerializeParallelOpsPass());
```

Comment:
Could we add some tests for this?

Reply:
what kind of test do you think we could add? Those passes are tested individually and there is an integration test for softmax already. The codegen pipeline tests would not include this part so I'm not sure what would be interested to test. What do you think?

Reply:
I guess it should be fine then. I was thinking about a test to make sure that no other pass mess up whatever this pass should be decomposing.

Reply:
You can try adding a pipeline test.

Reply:
Correct me if I am wrong, but I think the decompose softmax should probably be running elementwise operation fusion as well. You can probably also use the cost function of the fusion to incorporate rematerialization with the `decomposeSoftmax` code. That probably gives you the lowering you need?

Reply:
correct, I was planning to keep it for future clean up unless you think it makes things worse?